core/camel-api/src/main/java/org/apache/camel/spi/VaultConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.spi;
+
+/**
+ * Configuration for access to Vaults.
+ */
+public class VaultConfiguration {
+
+    @Metadata(secret = true)
```

Comment:
Does it make sense to have specific VaultConfiguration for each provider? I mean AWSVaultConfiguration, GCPVaultConfiguration etc

Reply:
Yeah maybe but lets start with this. Its a bit more tricky with different classes

Reply:
You would then have to go one level down on the configuation for aws / gcp etc 

VaultConfiguration cfg = ...

cfg.aws().getXXX

Or that we have to specify which sub class to get

AwsVaultConfiguration cfg = camelContext.getVaultConfiguration(AwsVaultConfiguration.class);

Reply:
I think it would be a good solution to subclass, in the end we shouldn't introduce new dependency, just some properties for connecting and defining the pointers. Btw LGTM. We could start from here.

Reply:
Okay I have a PR on the way with specific configuration per provider - AWS, Google, Azure etc